bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/DecimalType.java

```diff
@@ -41,19 +41,27 @@
     protected BigDecimal value;
 
     public DecimalType() {
-        this.value = BigDecimal.ZERO;
+        this(BigDecimal.ZERO);
     }
 
-    public DecimalType(BigDecimal value) {
-        this.value = value;
+    public DecimalType(Number value) {
+        this(bigDecimal(value));
     }
 
-    public DecimalType(long value) {
-        this.value = BigDecimal.valueOf(value);
+    private static BigDecimal bigDecimal(Number value) {
+        if (value.getClass() == QuantityType.class) {
```

Comment:
Why not use `instanceof` comparison?

```suggestion
        if (value instanceof QuantityType) {
```